mula/scheduler/server/handlers/queues.py

```diff
@@ -52,57 +53,53 @@ def __init__(self, api: fastapi.FastAPI, ctx: context.AppContext, s: dict[str, s
             description="Push an item to a queue",
         )
 
+    @exception_handler
     def list(self) -> Any:
         return [models.Queue(**s.queue.dict(include_pq=False)) for s in self.schedulers.copy().values()]
 
+    @exception_handler
     def get(self, queue_id: str) -> Any:
         s = self.schedulers.get(queue_id)
         if s is None:
             raise fastapi.HTTPException(
                 status_code=fastapi.status.HTTP_404_NOT_FOUND, detail="scheduler not found, by queue_id"
             )
 
-        q = s.queue
-        if q is None:
-            raise fastapi.HTTPException(status_code=fastapi.status.HTTP_404_NOT_FOUND, detail="queue not found")
-
-        return models.Queue(**q.dict())
+        return models.Queue(**s.queue.dict())
 
+    @exception_handler
     def pop(self, queue_id: str, filters: storage.filters.FilterRequest | None = None) -> Any:
         s = self.schedulers.get(queue_id)
         if s is None:
-            raise fastapi.HTTPException(status_code=fastapi.status.HTTP_404_NOT_FOUND, detail="queue not found")
+            raise fastapi.HTTPException(
+                status_code=fastapi.status.HTTP_404_NOT_FOUND, detail="scheduler not found, by queue id"
+            )
```

Comment:
You could move all 404's to a custom NotFoundException, and just raise that containing just the detail message. Catching that in the error handler reduced some boilerplate code.